protocol/x/stats/keeper/keeper.go

```diff
@@ -274,3 +280,75 @@ func (k Keeper) ExpireOldStats(ctx sdk.Context) {
 	metadata.TrailingEpoch += 1
 	k.SetStatsMetadata(ctx, metadata)
 }
+
+func (k Keeper) GetStakedAmount(ctx sdk.Context,
+	delegatorAddr string) big.Int {
+	startTime := time.Now()
+	var stakedAmount big.Int
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), []byte(types.CachedStakeAmountKeyPrefix))
+	bytes := store.Get([]byte(delegatorAddr))
+
+	// return cached value if it's not expired
+	if bytes != nil {
+		var cachedStakedAmount types.CachedStakeAmount
+		k.cdc.MustUnmarshal(bytes, &cachedStakedAmount)
+		if ctx.BlockTime().Unix()-int64(cachedStakedAmount.CachedAt) < int64(types.StakedAmountCacheDurationSeconds) {
+			stakedAmount.SetBytes(cachedStakedAmount.StakedAmount.BigInt().Bytes())
+			metrics.EmitTelemetryWithLabelsForExecMode(
+				ctx,
+				[]sdk.ExecMode{sdk.ExecModeFinalize},
+				metrics.IncrCounterWithLabels,
+				metrics.StatsGetStakedAmountCacheHit,
+				1,
+			)
+			metrics.EmitTelemetryWithLabelsForExecMode(
+				ctx,
+				[]sdk.ExecMode{sdk.ExecModeFinalize},
+				metrics.SetGaugeWithLabels,
+				metrics.StatsGetStakedAmountLatencyCacheHit,
+				float32(time.Since(startTime).Milliseconds()),
+			)
+			return stakedAmount
+		}
+	}
+
+	metrics.EmitTelemetryWithLabelsForExecMode(
+		ctx,
+		[]sdk.ExecMode{sdk.ExecModeFinalize},
```

Comment:
For these metrics and above, no need to guard by `ExecMode` since both `CheckTx` and `DeliverTx` are relevant